IGC/AdaptorCommon/LegalizeFunctionSignatures.cpp

```diff
@@ -548,7 +548,11 @@ void LegalizeFunctionSignatures::FixCallInstruction(Module& M, CallInst* callIns
         {
             // legal argument
             callArgs.push_back(arg);
+#if LLVM_VERSION_MAJOR >= 14
+            ArgAttrVec.push_back(PAL.getParamAttrs(opNum));
+#else
             ArgAttrVec.push_back(PAL.getParamAttributes(opNum));
+#endif
```

Comment:
Should be wrapped with

```cpp
    llvm::AttributeList getParamAttrs(llvm::AttributeList &AL, unsigned opNum)
    {
#if LLVM_VERSION_MAJOR >= 14
        return AL.getParamAttrs(opNum);
#else
        return AL.getParamAttributes(opNum);
#endif
    }
```

and then use in code:

```suggestion
            ArgAttrVec.push_back(IGCLLVM::getParamAttrs(PAL, opNum));
```

The original llvm code for this class is in `llvm/IR/Attributes.h` so the wrapper should be put in `/IGC/WrapperLLVM/include/llvmWrapper/IR/Attributes.h`.

This is one example. There should be similar wrappers for every situation where `#if LLVM_VERSION_MAJOR` is requried in code, and encapsulated away in the WrapperLLVM module under `IGCLLVM` namespace.